Dropdown/Dropdown.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Agate styled Header.
+ *
+ * @example
+ * <Dropdown
+ * 		defaultSelected={2}
+ *		inline
+ *		title="Dropdown"
+ * >
+ *   {['Option 1', 'Option 2', 'Option 3', 'Option 4']}
+ * </Dropdown>
+ *
+ * @module agate/Dropdown
+ * @exports Dropdown
+ */
+
+import React from 'react';
+import kind from '@enact/core/kind';
+import Changeable from '@enact/ui/Changeable';
+import Group from '@enact/ui/Group';
+import Icon from '@enact/agate/Icon';
+import Item from '@enact/agate/Item';
+import Toggleable from '@enact/ui/Toggleable';
+import Transition from '@enact/ui/Transition';
+import PropTypes from 'prop-types';
+import compose from 'ramda/src/compose';
+import Skinnable from '../Skinnable';
+import SpotlightContainerDecorator from '@enact/spotlight/SpotlightContainerDecorator';
+import Scroller from '@enact/ui/Scroller';
+import componentCss from './Dropdown.module.less';
+
+const ContainerDiv = SpotlightContainerDecorator({enterTo: 'last-focused'}, 'div');
+const isSelectedValid = ({children, selected}) => Array.isArray(children) && children[selected] != null;
+
+/**
+ * A stateless Dropdown component.
+ *
+ * @class DropdownBase
+ * @memberof agate/Dropdown
+ * @ui
+ * @public
+ */
+const DropdownBase = kind({
+	name: 'Dropdown',
+
+	propTypes: /** @lends agate/Dropdown.Dropdown.prototype */ {
+		/*
+		 * The selections for Dropdown
+		 *
+		 * @type {String[]|Array.<{key: (Number|String), children: (String|Component)}>}
+		 */
+		children: PropTypes.array,
+		/**
+		 * Customizes the component by mapping the supplied collection of CSS class names to the
+		 * corresponding internal elements and states of this component.
+		 *
+		 * @type {Object}
+		 * @public
+		 */
+		css: PropTypes.object,
+		/**
+		 * The animation direction of the `palette`.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		direction: PropTypes.string,
+		/*
+		 * Called when an item is selected.
+		 *
+		 * @type {Function}
+		 */
+		onSelect: PropTypes.func,
+		/*
+		 * Called when clicked on the dropdown to open.
+		 *
+		 * @type {Number}
+		 */
+
+		open: PropTypes.bool,
+		/*
+		 * Index of the selected item.
+		 *
+		 * @type {Number}
+		 */
+		selected: PropTypes.number
+	},
+
+	defaultProps: {
+		direction: 'down',
+		open: false
+	},
+
+	styles: {
+		css: componentCss,
+		className: 'Dropdown',
+		publicClassNames: true
+	},
+
+	computed: {
+		className: ({css, selected, styler}) => styler.append(selected && css.selected),
```

Comment:
I don't think we have a `selected` style. Although maybe we should have a selected style for the selected option in the dropdown list.